src/utils/utils.js

```diff
@@ -24,6 +24,9 @@ export const CTGROUPS = {
     BR: CTYPES.BR,
 };
 
+export const URL_REGEX = /((?:https?:\/\/)?(?:[a-z0-9-]{1,63}\.){1,2}[a-z]{2,15}(?:\/[^\s]*)?)/gi;
+export const URL_REGEX_WITH_INFOS = /((https?:\/\/)?([a-z0-9-]{1,63}\.){1,2}[a-z]{2,15}(\/[^\s]*)?)/gi;
```

Comment:
I assume it's a deliberate choice to only check http/https protocoles ?

Reply:
Yes.
I don't think we want to auto transform into link other protocols. do we ?

Reply:
It's safe to only deal with http/https because it covers most cases. If you have a specific case, create the user can still create a proper link through a modal.

Reply:
What I often see in editors is converting as soon as you have `\w+\.\w+` but that's how you end up with links whenever someone failed to put a space after a dot at the end of a sentence. This here is the other extreme where you _have_ to start you url with `http` for it to become a link, while in many cases you'd like to be able to simply type `www.site.com` or even `site.com`. Maybe the in-between solution is to cover `http`, `https` and `www.` ?

Reply:
Sorry, I was replying to the comments here... then I looked at the regex and saw the `?`s. So here, the following bit will be converted right?

> I'm typing some text.And forgot to put a space.

-> `<a href="https://text.And">text.And</a>`

Reply:
I think mailto is pretty common but indeed it's an entirely reasonable choice to only match http/https, I just wanted to make sure it wasn't an oversight.

Reply:
> Sorry, I was replying to the comments here... then I looked at the regex and saw the `?`s. So here, the following bit will be converted right?
> 
> > I'm typing some text.And forgot to put a space.
> 
> -> `<a href="https://text.And">text.And</a>`
> 
> (Not that it matters extremely but just to be clear that we're doing what we want to be doing ^^).

yes Indeed.